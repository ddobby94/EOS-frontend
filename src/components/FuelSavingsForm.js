import React from 'react';
import {func} from 'prop-types';
import {fuelSavings} from '../types';

const FuelSavingsForm = ({fuelSavings, onSaveClick, onChange}) => (
  <div>
    <h2>Fuel Savings Analysis</h2>
    <table>
      <tbody>

        <tr>
          <td><label htmlFor="milesDriven">Miles Driven</label></td>
        </tr>
        <tr>
          <td><label>Date Modified</label></td>
          <td>{fuelSavings.dateModified}</td>
        </tr>
      </tbody>
    </table>

    <hr/>
    <input type="submit" value="Save" onClick={onSaveClick}/>
  </div>
);

FuelSavingsForm.propTypes = {
  onSaveClick: func.isRequired,
  onChange: func.isRequired,
  fuelSavings: fuelSavings.isRequired
};

export default FuelSavingsForm;
