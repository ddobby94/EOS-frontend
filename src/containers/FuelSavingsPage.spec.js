import React from "react";
import { mount, shallow } from "enzyme";
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";
import { create } from "react-test-renderer";
import ConnectedFuelSavingsPage, { FuelSavingsPage } from "./FuelSavingsPage";
import FuelSavingsForm from "../components/FuelSavingsForm";

const initialState = {
  fuelSavings: {
    newMpg: '',
    tradeMpg: '',
    newPpg: '',
    tradePpg: '',
    milesDriven: '',
    milesDrivenTimeframe: 'week',
    displayResults: false,
    dateModified: null,
    necessaryDataIsProvidedToCalculateSavings: false,
    savings: {
      monthly: 0,
      annual: 0,
      threeYear: 0
    }
  }
};

describe("<FuelSavingsPage />", () => {
  const actions = {
    saveFuelSavings: jest.fn(),
    calculateFuelSavings: jest.fn()
  };

  it("should contain <FuelSavingsForm />", () => {
    const wrapper = shallow(
      <FuelSavingsPage
        actions={actions}
        fuelSavings={initialState.fuelSavings}
      />
    );

    expect(wrapper.find(FuelSavingsForm).length).toEqual(1);
  });

  it("calls saveFuelSavings upon clicking save", () => {
    const wrapper = mount(
      <FuelSavingsPage
        actions={actions}
        fuelSavings={initialState.fuelSavings}
      />
    );

    const save = wrapper.find('input[type="submit"]');
    save.simulate("click");

    expect(actions.saveFuelSavings).toHaveBeenCalledWith(
      initialState.fuelSavings
    );
  });


  it("should match snapshot", () => {
    const store = configureMockStore()(initialState);
    const component = create(
      <Provider store={store}>
        <ConnectedFuelSavingsPage />
      </Provider>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
