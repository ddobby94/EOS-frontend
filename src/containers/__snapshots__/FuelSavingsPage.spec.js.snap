// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FuelSavingsPage /> should match snapshot 1`] = `
Array [
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Primary
    </span>
  </button>,
  <div>
    <h2>
      Fuel Savings Analysis
    </h2>
    <table>
      <tbody>
        <tr>
          <td>
            <label
              htmlFor="milesDriven"
            >
              Miles Driven
            </label>
          </td>
        </tr>
        <tr>
          <td>
            <label>
              Date Modified
            </label>
          </td>
          <td />
        </tr>
      </tbody>
    </table>
    <hr />
    <input
      onClick={[Function]}
      type="submit"
      value="Save"
    />
  </div>,
]
`;
